### ISSUE TITLE
Console command can break setup

### ISSUE THREAD

**Issue by FreekVandeursen** [magento/magento2#16173]
### Preconditions
1. Magento 2.2.4
2. Empty database

### Steps to reproduce
1. Create a module, containing a console command
2. Use ProductRepositoryInterface in console command constructor
3. run bin/magento setup:install <install parameters>

### Expected result
1. Setup completing succesfully

### Actual result
1. Database exception, claiming table 'website' does not exist

### Analysis
The root cause seems to be in the `Magento\Setup\Model\ObjectManagerProvider` class, which calls `createCliCommands` in it's `get` function. This will break setup when there are any console commands that depend on the database to exist already. I have seen this happen with the ProductRepository for example.

What is the reason for this createCliCommands call? If I just disable that call, the setup will complete succesfully.
It also seems strange that this call is only present in the CLI SAPI, which means functionality is different between CLI or web installation.

### Example
```
<?php
namespace Test\Test\Console\Command;

use Magento\Catalog\Api\ProductRepositoryInterface;
use Symfony\Component\Console\Command\Command;

class TestCommand extends Command
{

    /**
     * @var ProductRepositoryInterface
     */
    private $productRepository;

    public function __construct(
        ProductRepositoryInterface $productRepository,
        string $name = null
    )
    {
        parent::__construct($name);
        $this->productRepository = $productRepository;
    }

    protected function configure()
    {
        $this->setName('test:test')
            ->setDescription('Test command');
        parent::configure();
    }
}
```

**Comment by will-b** [comment]
@FreekVandeursen Think a proxy should be used instead of the product repository interface to avoid a dependency trying to read the 'website' table.
https://devdocs.magento.com/guides/v2.2/extension-dev-guide/proxies.html
Will see if I can find an example tomorrow.

**Comment by FreekVandeursen** [comment]
@will-b I agree with you that there are ways of working around this problem. I wonder though if the current behaviour is intended. I think there are 2 ways of looking at this:
1. Should just including a repository in a constructor lead to a exception on a system which is not installed yet?
2. Why does the setup ObjectManager try to generate classes for all CLI commands, even those who are not yet available before installation?

**Comment by engcom-backlog-nickolas** [comment]
@FreekVandeursen, thank you for your report. As your module dependent on catalog module, you need to specify dependency for you module: https://devdocs.magento.com/guides/v2.2/extension-dev-guide/build/module-load-order.html

**Comment by FreekVandeursen** [comment]
@engcom-backlog-nickolas I did specify a dependency in my module.xml, that doesn't solve the problem.

**Comment by engcom-backlog-nickolas** [comment]
@FreekVandeursen , thank you for your feedback.
We've acknowledged the issue and added to our backlog.

**Comment by Bartlomiejsz** [comment]
@FreekVandeursen @engcom-backlog-nazar possible that it was duplicated in #21752 which is closed due to #21693

**Comment by engcom-Charlie** [comment]
Hello @FreekVandeursen 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates `2.3-develop` branch by provided steps.

We are closing this issue due to _branch `2.2-develop` was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3._
 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)
